src/transformers/trainer.py

```diff
@@ -233,6 +242,83 @@ def __init__(
+                The dataset loaded and preprocessed via the :obj:`nlp` library.
+            data_collator (:obj:`DataCollator`, `optional`, defaults to :func:`~transformers.default_data_collator`):
+                The function to use to form a batch from a list of elements from the :obj:`dataset`.
+            metrics (:obj:`nlp.Metric` or :obj:`List[nlp.Metric]`, `optional`):
+                One or several metrics loaded via the :obj:`nlp` library.
+            final_activation (:obj:`str` or :class:`~transformers.trainer_utils.FinalActivation`, `optional`):
+                The final activation to apply to the model output before feeding them to the :obj:`metrics`.
+            tb_writer (:obj:`SummaryWriter`, `optional`):
+                Object to write to TensorBoard.
+            optimizers (:obj:`Tuple[torch.optim.Optimizer, torch.optim.lr_scheduler.LambdaLR`, `optional`):
+                A tuple containing the optimizer and the scheduler to use. Will default to an instance of
+                :class:`~transformers.AdamW` on your model and a scheduler given by
+                :func:`~transformers.get_linear_schedule_with_warmup` controlled by :obj:`args`.
+
+        Examples:
+
+            >>> from nlp import load_dataset, load_metric
+            >>> from transformers import AutoModelForSequenceClassification, AutoTokenizer, Trainer, TrainingArguments
+            >>> dataset = load_dataset('glue', 'sst2')
+            >>> metric = load_metric('glue', 'sst2')
+
+            >>> model_name = "distilbert-base-uncased-finetuned-sst-2-english"
+            >>> tokenizer = AutoTokenizer.from_pretrained(model_name)
+            >>> model = AutoModelForSequenceClassification.from_pretrained(model_name)
+
+            >>> encoded_dataset = dataset.map(lambda examples: tokenizer(examples['sentence'], padding=True), batched=True)
+            >>> args = TrainingArguments(output_dir = "test")
+
+            >>> trainer = Trainer.from_nlp_dataset(model, args, encoded_dataset, metrics=metric, final_activation="argmax")
+            >>> trainer.evaluate()
+            {'eval_loss': 0.3902028881816018, 'eval_accuracy': 0.9105504587155964, 'step': 0}
+        """
+        assert is_nlp_available(), "This method requires the nlp library: `pip install nlp`."
+        if metrics is not None:
+            compute_metrics = ComputeNLPMetrics(metrics, activation=final_activation)
+
+        # Inspect model forward signature to keep only the arguments it accepts.
+        signature = inspect.signature(model.forward)
+        signature_columns = list(signature.parameters.keys())
+        # Labels may be named label or label_ids, the default data collator handles that.
+        signature_columns += ["label", "label_ids"]
+        dataset_columns = dataset[list(dataset.keys())[0]].column_names
+        columns = [k for k in signature_columns if k in dataset_columns]
+        dataset.set_format(columns=columns)
```

Comment:
Could we also print a `logger.warning` if some values are skipped?

Reply:
Did an info instead of a warning since it's what the API is supposed to do and users are scared of warnings. Also properly documented the API is doing that.